applications/teletype/src/templates/teletype_customer_update.erl

```diff
@@ -0,0 +1,157 @@
+                                           ,{'text', ?TEMPLATE_TEXT}
+                                           ,{'html', ?TEMPLATE_HTML}
+                                           ,{'subject', ?TEMPLATE_SUBJECT}
+                                           ,{'category', ?TEMPLATE_CATEGORY}
+                                           ,{'friendly_name', ?TEMPLATE_NAME}
+                                           ,{'to', ?TEMPLATE_TO}
+                                           ,{'from', ?TEMPLATE_FROM}
+                                           ,{'cc', ?TEMPLATE_CC}
+                                           ,{'bcc', ?TEMPLATE_BCC}
+                                           ,{'reply_to', ?TEMPLATE_REPLY_TO}
+                                          ]).
+
+-spec handle_req(wh_json:object(), wh_proplist()) -> 'ok'.
+handle_req(JObj, _Props) ->
+    'true' = wapi_notifications:customer_update_v(JObj),
+    DataJObj = wh_json:normalize(JObj),
+    AccountId = wh_json:get_value(<<"account_id">>, DataJObj),
+    case teletype_util:is_notice_enabled(AccountId, JObj, ?TEMPLATE_ID) of
+        'false' -> lager:debug("notification handling not configured for this account");
+        'true' -> process_req(DataJObj)
+    end.
+
+-spec process_req(wh_json:object()) -> 'ok'.
+process_req(DataJObj) ->
+    SenderId = wh_json:get_value(<<"account_id">>, DataJObj),
+    ViewOpts = [{'startkey', [SenderId]}
+               ,{'endkey', [SenderId, wh_json:new()]}
+               ],
+    case couch_mgr:get_results(?WH_ACCOUNTS_DB, <<"accounts/listing_by_children">>, ViewOpts) of
+        {'ok', Accounts} ->
+            process_accounts(Accounts, DataJObj);
+        {'error', _Reason} = E ->
+            lager:info("failed to load children. error: ~p", [E]),
+            E
+    end.
+
+process_accounts(Accounts, DataJObj) ->
+    case wh_json:get_value(<<"recipient_id">>, DataJObj) of
+        'undefined' ->
+            [process_account(wh_json:get_value(<<"value">>, Account), DataJObj) || Account <- Accounts];
+        RecipientId ->
+            [process_account(wh_json:get_value(<<"value">>, Account), DataJObj) || Account <- Accounts, wh_json:get_value(<<"id">>, Account) == RecipientId]
+    end.
+
+process_account(AccountJObj, DataJObj) ->
+    AccountId = wh_json:get_value(<<"id">>, AccountJObj),
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    case couch_mgr:get_results(AccountDb,<<"users/crossbar_listing">>,[]) of
+        {'ok', Users} ->
+            Templates = teletype_templates:fetch(?TEMPLATE_ID, DataJObj),
+            select_users_to_update(lists:map(fun(User) -> wh_json:get_value(<<"value">>, User) end, Users), DataJObj, Templates);
+        {'error', _Reason} = E ->
+            lager:info("failed to load users from: ~p", [AccountDb]),
+            E
+    end.
+
+select_users_to_update(Users, DataJObj, Templates) ->
+    case wh_json:get_value(<<"user_type">>, DataJObj) of
```

Comment:
Similarly to `recipient_id`, if you know a particular `user_id` to use, go for it; otherwise load the list of users and go from there.